### ISSUE TITLE
Literate config not working with org headings

### ISSUE THREAD

**Issue by justyns** [hlissner/doom-emacs#3696]
**What did you expect to happen?**

I created a simple `~/.doom.d/config.org` file to test the literate config module, and expected it to generate `~/.doom.d/config.el` when running `doom sync`.


**What actually happened?**

If my `config.org` contains elisp code blocks, but no org headings - everything seems to work fine.  However, if I include any org headings at all - I get this error when running `doom sync`:

```
> Compiling your literate config...
Wrong type argument: stringp, nil
```


**Additional details:**
- Example of a config.org that does not error out:
``` org
Test.  I can put random data here outside the src block.

#+BEGIN_SRC emacs-lisp
;; test 0
#+END_SRC
```

- Example of a config.org that causes the error:

``` org
blah

* Top level heading
** Second level heading
And then some text goes here..

#+BEGIN_SRC emacs-lisp
;; test 0
#+END_SRC

```

- I'm not sure if it matters, but I also get the same error when trying to save `config.org`.
- This is a backtrace from `doom sync`:

<details><pre>
➜  doom.d git:(master) ✗ DEBUG=1 doom sync                                                                                                                                                                                                                         git:(master|✚1…
DOOM Initializing Doom CLI
DOOM Initializing straight
DOOM Initializing recipes
DOOM Installing core packages
> Compiling your literate config...
  - Debugger entered--Lisp error: (wrong-type-argument stringp nil)
    looking-at(nil)
    org-heading-components()
    org-in-commented-heading-p()
    org-babel-tangle-collect-blocks(nil nil)
    org-babel-tangle(nil "~/.doom.d/config.el")
    (let ((buffer-file-name backup)) (insert-file-contents org) (org-export-expand-include-keyword) (org-babel-tangle nil dest))
    (save-current-buffer (set-buffer temp-buffer) (let ((buffer-file-name backup)) (insert-file-contents org) (org-export-expand-include-keyword) (org-babel-tangle nil dest)))
    (prog1 (save-current-buffer (set-buffer temp-buffer) (let ((buffer-file-name backup)) (insert-file-contents org) (org-export-expand-include-keyword) (org-babel-tangle nil dest))) (save-current-buffer (set-buffer temp-buffer) (write-region nil nil temp-file nil 0)))
    (unwind-protect (prog1 (save-current-buffer (set-buffer temp-buffer) (let ((buffer-file-name backup)) (insert-file-contents org) (org-export-expand-include-keyword) (org-babel-tangle nil dest))) (save-current-buffer (set-buffer temp-buffer) (write-region nil nil temp-file nil 0))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer)))
    (let ((temp-file backup) (temp-buffer (get-buffer-create (generate-new-buffer-name " *temp file*")))) (unwind-protect (prog1 (save-current-buffer (set-buffer temp-buffer) (let ((buffer-file-name backup)) (insert-file-contents org) (org-export-expand-include-keyword) (org-babel-tangle nil dest))) (save-current-buffer (set-buffer temp-buffer) (write-region nil nil temp-file nil 0))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer))))
    (progn (let ((temp-file backup) (temp-buffer (get-buffer-create (generate-new-buffer-name " *temp file*")))) (unwind-protect (prog1 (save-current-buffer (set-buffer temp-buffer) (let ((buffer-file-name backup)) (insert-file-contents org) (org-export-expand-include-keyword) (org-babel-tangle nil dest))) (save-current-buffer (set-buffer temp-buffer) (write-region nil nil temp-file nil 0))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer)))) t)
    (let ((org-mode-hook nil)) (progn (let ((temp-file backup) (temp-buffer (get-buffer-create (generate-new-buffer-name " *temp file*")))) (unwind-protect (prog1 (save-current-buffer (set-buffer temp-buffer) (let (...) (insert-file-contents org) (org-export-expand-include-keyword) (org-babel-tangle nil dest))) (save-current-buffer (set-buffer temp-buffer) (write-region nil nil temp-file nil 0))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer)))) t))
    (progn (fset (function message) vnew) (ignore message) (let ((org-mode-hook nil)) (progn (let ((temp-file backup) (temp-buffer (get-buffer-create (generate-new-buffer-name " *temp file*")))) (unwind-protect (prog1 (save-current-buffer (set-buffer temp-buffer) (let ... ... ... ...)) (save-current-buffer (set-buffer temp-buffer) (write-region nil nil temp-file nil 0))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer)))) t)))
    (unwind-protect (progn (fset (function message) vnew) (ignore message) (let ((org-mode-hook nil)) (progn (let ((temp-file backup) (temp-buffer (get-buffer-create ...))) (unwind-protect (prog1 (save-current-buffer ... ...) (save-current-buffer ... ...)) (and (buffer-name temp-buffer) (kill-buffer temp-buffer)))) t))) (fset (function message) old))
    (let* ((vnew (function (lambda (msg &rest args) (if msg (progn (doom--print ...)))))) (old (symbol-function (function message)))) (unwind-protect (progn (fset (function message) vnew) (ignore message) (let ((org-mode-hook nil)) (progn (let ((temp-file backup) (temp-buffer ...)) (unwind-protect (prog1 ... ...) (and ... ...))) t))) (fset (function message) old)))
    (let ((message (symbol-function (function message)))) (let* ((vnew (function (lambda (msg &rest args) (if msg (progn ...))))) (old (symbol-function (function message)))) (unwind-protect (progn (fset (function message) vnew) (ignore message) (let ((org-mode-hook nil)) (progn (let (... ...) (unwind-protect ... ...)) t))) (fset (function message) old))))
    (and (require (quote ox)) (require (quote ob-tangle)) (let ((message (symbol-function (function message)))) (let* ((vnew (function (lambda (msg &rest args) (if msg ...)))) (old (symbol-function (function message)))) (unwind-protect (progn (fset (function message) vnew) (ignore message) (let ((org-mode-hook nil)) (progn (let ... ...) t))) (fset (function message) old)))) (let ((temp-file +literate-config-cache-file) (temp-buffer (get-buffer-create (generate-new-buffer-name " *temp file*")))) (unwind-protect (prog1 (save-current-buffer (set-buffer temp-buffer)) (save-current-buffer (set-buffer temp-buffer) (write-region nil nil temp-file nil 0))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer)))))
    (unwind-protect (and (require (quote ox)) (require (quote ob-tangle)) (let ((message (symbol-function (function message)))) (let* ((vnew (function (lambda ... ...))) (old (symbol-function (function message)))) (unwind-protect (progn (fset (function message) vnew) (ignore message) (let (...) (progn ... t))) (fset (function message) old)))) (let ((temp-file +literate-config-cache-file) (temp-buffer (get-buffer-create (generate-new-buffer-name " *temp file*")))) (unwind-protect (prog1 (save-current-buffer (set-buffer temp-buffer)) (save-current-buffer (set-buffer temp-buffer) (write-region nil nil temp-file nil 0))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer))))) (condition-case nil (progn (delete-file backup)) (error nil)))
    (let* ((default-directory doom-private-dir) (org (expand-file-name +literate-config-file)) (dest (concat (file-name-sans-extension +literate-config-file) ".el")) (backup (make-temp-file (concat (file-name-nondirectory org) ".")))) (unwind-protect (and (require (quote ox)) (require (quote ob-tangle)) (let ((message (symbol-function (function message)))) (let* ((vnew (function ...)) (old (symbol-function ...))) (unwind-protect (progn (fset ... vnew) (ignore message) (let ... ...)) (fset (function message) old)))) (let ((temp-file +literate-config-cache-file) (temp-buffer (get-buffer-create (generate-new-buffer-name " *temp file*")))) (unwind-protect (prog1 (save-current-buffer (set-buffer temp-buffer)) (save-current-buffer (set-buffer temp-buffer) (write-region nil nil temp-file nil 0))) (and (buffer-name temp-buffer) (kill-buffer temp-buffer))))) (condition-case nil (progn (delete-file backup)) (error nil))))
    (let ((doom-output-indent (+ doom-output-indent-increment doom-output-indent))) (let* ((default-directory doom-private-dir) (org (expand-file-name +literate-config-file)) (dest (concat (file-name-sans-extension +literate-config-file) ".el")) (backup (make-temp-file (concat (file-name-nondirectory org) ".")))) (unwind-protect (and (require (quote ox)) (require (quote ob-tangle)) (let ((message (symbol-function ...))) (let* ((vnew ...) (old ...)) (unwind-protect (progn ... ... ...) (fset ... old)))) (let ((temp-file +literate-config-cache-file) (temp-buffer (get-buffer-create ...))) (unwind-protect (prog1 (save-current-buffer ...) (save-current-buffer ... ...)) (and (buffer-name temp-buffer) (kill-buffer temp-buffer))))) (condition-case nil (progn (delete-file backup)) (error nil)))))
    +literate-tangle-h()
    run-hooks(doom-cli-pre-hook)
    (let ((help-p (cdr (assq (quote help-p) --alist--))) (debug-p (cdr (assq (quote debug-p) --alist--))) (yes-p (cdr (assq (quote yes-p) --alist--))) (command (cdr (assq (quote command) --alist--))) (args (cdr (assq (quote args) --alist--)))) :bare t (if debug-p (progn (setenv "DEBUG" "1") (setq doom-debug-p t) (print! (info "Debug mode on")))) (if yes-p (progn (setenv "YES" "1") (setq doom-auto-accept t) (print! (info "Auto-yes on")))) (if help-p (progn (if command (progn (setq args (cons command args)))) (setq command "help"))) (if (equal (user-real-uid) 0) (progn (print! (concat "WARNING: This script is running as root. This likely wasn't intentional, and\n" "is unnecessary to use this script. This will cause file permissions errors\n" "later if you use this Doom installation on a non-root account.\n")) (if (or doom-auto-accept (y-or-n-p "Continue anyway?")) nil (user-error "Aborted")))) (load! doom-module-init-file doom-private-dir t) (maphash (doom-module-loader doom-cli-file) doom-modules) (load! doom-cli-file doom-private-dir t) (run-hooks (quote doom-cli-pre-hook)) (let (print-level print-gensym print-length) (condition-case e (if (null command) (doom-cli-execute "help") (let ((start-time (current-time))) (and (doom-cli-execute command args) (progn (run-hooks ...) t) (print! (success "Finished! (%.4fs)") (float-time ...))))) (user-error (print! (error "%s\n") (error-message-string e)) (print! (yellow "See 'doom help %s' for documentation on this command.") (car args)) (error "")) ((debug error) (print! (error "There was an unexpected error:")) (print-group! (print! "%s %s" (bold "Type:") (car e)) (print! (bold "Message:")) (print-group! (print! "%s" (get ... ...))) (print! (bold "Data:")) (print-group! (if (cdr e) (let ... ...) (print! "n/a"))) (if (and (and ... straight-process-buffer) (string-match-p ... ...)) (progn (print! ...) (print-group! ...)))) (if debug-on-error nil (terpri) (print! (concat "Run the command again with the -d (or --debug) switch to enable debug\n" "mode and (hopefully) generate a backtrace from this error:\n" "\n  %s\n\n" "If you file a bug report, please include it!") (string-join (append ... args) " ")) (error ""))))))
    (lambda (--alist--) (ignore --alist--) (let ((help-p (cdr (assq (quote help-p) --alist--))) (debug-p (cdr (assq (quote debug-p) --alist--))) (yes-p (cdr (assq (quote yes-p) --alist--))) (command (cdr (assq (quote command) --alist--))) (args (cdr (assq (quote args) --alist--)))) :bare t (if debug-p (progn (setenv "DEBUG" "1") (setq doom-debug-p t) (print! (info "Debug mode on")))) (if yes-p (progn (setenv "YES" "1") (setq doom-auto-accept t) (print! (info "Auto-yes on")))) (if help-p (progn (if command (progn (setq args (cons command args)))) (setq command "help"))) (if (equal (user-real-uid) 0) (progn (print! (concat "WARNING: This script is running as root. This likely wasn't intentional, and\n" "is unnecessary to use this script. This will cause file permissions errors\n" "later if you use this Doom installation on a non-root account.\n")) (if (or doom-auto-accept (y-or-n-p "Continue anyway?")) nil (user-error "Aborted")))) (load! doom-module-init-file doom-private-dir t) (maphash (doom-module-loader doom-cli-file) doom-modules) (load! doom-cli-file doom-private-dir t) (run-hooks (quote doom-cli-pre-hook)) (let (print-level print-gensym print-length) (condition-case e (if (null command) (doom-cli-execute "help") (let ((start-time ...)) (and (doom-cli-execute command args) (progn ... t) (print! ... ...)))) (user-error (print! (error "%s\n") (error-message-string e)) (print! (yellow "See 'doom help %s' for documentation on this command.") (car args)) (error "")) ((debug error) (print! (error "There was an unexpected error:")) (print-group! (print! "%s %s" (bold "Type:") (car e)) (print! (bold "Message:")) (print-group! (print! "%s" ...)) (print! (bold "Data:")) (print-group! (if ... ... ...)) (if (and ... ...) (progn ... ...))) (if debug-on-error nil (terpri) (print! (concat "Run the command again with the -d (or --debug) switch to enable debug\n" "mode and (hopefully) generate a backtrace from this error:\n" "\n  %s\n\n" "If you file a bug report, please include it!") (string-join ... " ")) (error "")))))))(((command . "sync")))
    funcall((lambda (--alist--) (ignore --alist--) (let ((help-p (cdr (assq (quote help-p) --alist--))) (debug-p (cdr (assq (quote debug-p) --alist--))) (yes-p (cdr (assq (quote yes-p) --alist--))) (command (cdr (assq (quote command) --alist--))) (args (cdr (assq (quote args) --alist--)))) :bare t (if debug-p (progn (setenv "DEBUG" "1") (setq doom-debug-p t) (print! (info "Debug mode on")))) (if yes-p (progn (setenv "YES" "1") (setq doom-auto-accept t) (print! (info "Auto-yes on")))) (if help-p (progn (if command (progn (setq args ...))) (setq command "help"))) (if (equal (user-real-uid) 0) (progn (print! (concat "WARNING: This script is running as root. This likely wasn't intentional, and\n" "is unnecessary to use this script. This will cause file permissions errors\n" "later if you use this Doom installation on a non-root account.\n")) (if (or doom-auto-accept (y-or-n-p "Continue anyway?")) nil (user-error "Aborted")))) (load! doom-module-init-file doom-private-dir t) (maphash (doom-module-loader doom-cli-file) doom-modules) (load! doom-cli-file doom-private-dir t) (run-hooks (quote doom-cli-pre-hook)) (let (print-level print-gensym print-length) (condition-case e (if (null command) (doom-cli-execute "help") (let (...) (and ... ... ...))) (user-error (print! (error "%s\n") (error-message-string e)) (print! (yellow "See 'doom help %s' for documentation on this command.") (car args)) (error "")) ((debug error) (print! (error "There was an unexpected error:")) (print-group! (print! "%s %s" ... ...) (print! ...) (print-group! ...) (print! ...) (print-group! ...) (if ... ...)) (if debug-on-error nil (terpri) (print! ... ...) (error ""))))))) ((command . "sync")))
    (if cli (funcall (progn (or (and (memq (type-of cli) cl-struct-doom-cli-tags) t) (signal (quote wrong-type-argument) (list (quote doom-cli) cli))) (aref cli 7)) (doom--cli-process cli (remq nil args))) (user-error "Couldn't find any %S command" command))
    (let* ((cli (and t (doom-cli-get command)))) (if cli (funcall (progn (or (and (memq (type-of cli) cl-struct-doom-cli-tags) t) (signal (quote wrong-type-argument) (list (quote doom-cli) cli))) (aref cli 7)) (doom--cli-process cli (remq nil args))) (user-error "Couldn't find any %S command" command)))
    doom-cli-execute(:main ("sync"))
    (let* ((loaddir (file-name-directory (file-truename load-file-name))) (emacsdir (getenv "EMACSDIR")) (user-emacs-directory (abbreviate-file-name (if emacsdir (file-name-as-directory emacsdir) (expand-file-name "../" loaddir))))) (load (expand-file-name "core/core.el" user-emacs-directory) nil t) (quiet! (if (> emacs-major-version 26) (progn (require (quote cl)))) (load "site-start" t t)) (doom-log "Initializing Doom CLI") (require (quote core-cli)) (defcli! :main ((help-p ["-h" "--help"] "Same as help command") (debug-p ["-d" "--debug"] "Turns on doom-debug-p (and debug-on-error)") (yes-p ["-y" "--yes"] "Auto-accept all confirmation prompts") &optional command &rest args) "A command line interface for managing Doom Emacs.\n\nIncludes package management, diagnostics, unit tests, and byte-compilation.\n\nThis tool also makes it trivial to launch Emacs out of a different folder or\nwith a different private module." :bare t (if debug-p (progn (setenv "DEBUG" "1") (setq doom-debug-p t) (print! (info "Debug mode on")))) (if yes-p (progn (setenv "YES" "1") (setq doom-auto-accept t) (print! (info "Auto-yes on")))) (if help-p (progn (if command (progn (setq args (cons command args)))) (setq command "help"))) (if (equal (user-real-uid) 0) (progn (print! (concat "WARNING: This script is running as root. This likely wasn't intentional, and\n" "is unnecessary to use this script. This will cause file permissions errors\n" "later if you use this Doom installation on a non-root account.\n")) (if (or doom-auto-accept (y-or-n-p "Continue anyway?")) nil (user-error "Aborted")))) (load! doom-module-init-file doom-private-dir t) (maphash (doom-module-loader doom-cli-file) doom-modules) (load! doom-cli-file doom-private-dir t) (run-hooks (quote doom-cli-pre-hook)) (let (print-level print-gensym print-length) (condition-case e (if (null command) (doom-cli-execute "help") (let ((start-time ...)) (and (doom-cli-execute command args) (progn ... t) (print! ... ...)))) (user-error (print! (error "%s\n") (error-message-string e)) (print! (yellow "See 'doom help %s' for documentation on this command.") (car args)) (error "")) ((debug error) (print! (error "There was an unexpected error:")) (print-group! (print! "%s %s" (bold "Type:") (car e)) (print! (bold "Message:")) (print-group! (print! "%s" ...)) (print! (bold "Data:")) (print-group! (if ... ... ...)) (if (and ... ...) (progn ... ...))) (if debug-on-error nil (terpri) (print! (concat "Run the command again with the -d (or --debug) switch to enable debug\n" "mode and (hopefully) generate a backtrace from this error:\n" "\n  %s\n\n" "If you file a bug report, please include it!") (string-join ... " ")) (error "")))))) (doom-cli-execute :main (cdr (member "--" argv))) (setq argv nil))
    eval-buffer(#<buffer  *load*> nil "/home/justyns/.emacs.d/bin/doom" nil t)  ; Reading at buffer position 5414
    load-with-code-conversion("/home/justyns/.emacs.d/bin/doom" "/home/justyns/.emacs.d/bin/doom" nil t)
    load("/home/justyns/.emacs.d/bin/doom" nil t t)
    command-line-1(("-scriptload" "/home/justyns/.emacs.d/bin/doom" "--" "sync"))
    command-line()
    normal-top-level()

</pre></details>


**Steps to reproduce:**
1. Enable `literate` doom module
2. Try the two example `config.org` files above
3. Run `doom sync` with the one containing org headings


**System information:**
<details><pre>
((emacs
  (version . "26.2")
  (features . "XPM JPEG TIFF GIF PNG SOUND GPM DBUS GSETTINGS GLIB NOTIFY GNUTLS LIBXML2 FREETYPE XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM THREADS")
  (build . "Aug 24, 2019")
  (buildopts "")
  (windowsys . batch)
  (daemonp))
 (doom
  (version . "2.0.9")
  (build . "grafted, HEAD -> develop, origin/develop, origin/HEAD 01c5c35 2020-08-04 18:04:15 -0400")
  (dir . "~/.doom.d/"))
 (system
  (type . gnu/linux)
  (config . "x86_64-pc-linux-gnu")
  (shell . "/usr/bin/zsh")
  (uname . "Linux 3.10.0-1062.12.1.el7.x86_64 #1 SMP Tue Feb 4 23:02:59 UTC 2020 x86_64")
  (path "~/.rbenv/shims" "~/.krew/bin" "~/.rbenv/shims" "~/.krew/bin" "/usr/local/bin" "/usr/bin" "~/bin" "/usr/local/sbin" "/usr/sbin" "~/.fzf/bin" "~/.rbenv/bin" "~/bin" "~/.rbenv/bin" "~/.local/bin" "~/bin" "~/.local/bin" "~/.rbenv/bin" "~/.emacs.d/bin" "~/go/bin" "/usr/local/libexec/emacs/26.2/x86_64-pc-linux-gnu"))
 (config
  (envfile . envvar-file)
  (elc-files . 0)
  (modules :completion company ivy :ui doom doom-dashboard doom-quit hl-todo modeline ophints (popup +defaults) vc-gutter vi-tilde-fringe workspaces :editor (evil +everywhere) file-templates fold format snippets :emacs dired electric undo vc :term vterm :checkers syntax :tools editorconfig (eval +overlay) lookup lsp magit terraform :lang emacs-lisp markdown org (python +lsp) sh yaml :config literate (default +bindings +smartparens))
  (packages "n/a")
  (unpin "n/a")
  (elpa "n/a")))
</pre></details>

**Comment by hlissner** [comment]
This is a duplicate of #3688 and was fixed in 00a8f75. Please run `doom upgrade`.

**Comment by justyns** [comment]
Oops, I started creating this issue last night after a fresh install and did not try upgrading before submitting tonight.

Can confirm this is fixed after `doom upgrade`.  Sorry for the noise, thanks @hlissner!
